pages/api/check-status.ts

```diff
@@ -81,24 +81,36 @@ export default async function handler(
   req: NextApiRequest,
   res: NextApiResponse<CheckStatusReponse | string>
 ) {
-  if (req.method !== 'POST') {
-    return res.status(405).send(`Invalid request method ${req.method}`)
+  if (req.method !== 'GET') {
+    res.status(405).send(`Invalid request method ${req.method}`)
+    return
   }
 
-  const body = JSON.parse(req.body) as CheckStatusRequestBody
+  const { searchParams } = new URL(req.url ?? '', `http://${req.headers.host}`)
+  const checkStatusRequest: CheckStatusRequest = {
+    birthDate: searchParams.get('birthDate') ?? '',
+    esrf: searchParams.get('esrf') ?? '',
+    givenName: searchParams.get('givenName') ?? '',
+    surname: searchParams.get('surname') ?? '',
+  }
```

Comment:
This is what feels "janky" and error prone to a simple typo and why I didn't want to pursue this change atm.

Reply:
I don't think it's "janky" and error prone. @gregory-j-baker what's your point of view?

Reply:
My first thought when I saw this was that there is definitely some jank here.. the framework should provide this functionality, and having to do it by hand by wrapping `req.url` in `new URL()` just looks weird.

And resolving the query params using strings is definitely prone to typos; but even if this were somehow strongly typed, those types would end up just being hand-typed types (man that's a lot of _type_). My point is that resolving query params is prone to error no matter how you do it; someone is going to have to type these strings out at some point, whether it be in a `searchParams.get()` or a `interface CheckStatusRequestType { .. }`.

I don't have a strong point of view either way.. this code is ugly (not your fault), but it works. It would be nice if there was a cleaner way to do it.

Reply:
That is exactly my feeling of it too - but I don't think it's worth blocking the PR over it. Maybe this is something we could build a little custom function for later to handle in a slightly "cleaner", more predictable fashion - even through in some logging ;).